lms/djangoapps/program_enrollments/models.py

```diff
@@ -136,6 +137,18 @@ class Meta(object):
     def is_active(self):
         return self.status == ProgramCourseEnrollmentStatuses.ACTIVE
 
+    @property
+    def course_staff(self):
+        role_assignment_exists = CourseAccessRoleAssignment.objects.filter(
+            enrollment=self,
+            role=ProgramCourseEnrollmentRoles.COURSE_STAFF,
+        ).exists()
+        if role_assignment_exists:
+            return True
+        elif self.program_enrollment.user:
```

Comment:
could probably do this check first so we only need to query CourseAccessRoleAssignment if needed.